### DISCUSSION TITLE
Ideas to think about in very long future in next major version

### DISCUSSION THREAD

**Discussion by nitbharambe** [PowerGridModel/.github#39] [Ideas]
See also https://github.com/PowerGridModel/power-grid-model/milestone/3

If we should do ever to have a major version bump introducing breaking changes, we can include the following:

- Capitalized enums as thats PEP standard
- Mandatory keyword arguments to existing functions
- ~Removing literal string~ done

This is not in planning. It is just a collection of ideas to not forget them.

**Comment by figueroa1395** [comment]
Refactor iterator usage in general to "comply" with ranges or make use of it for more readability and perhaps better performance. For examples `update_component` functions in `main_model_impl.hpp`.

**Reply by mgovers** [reply]
i agree that this would be great, but it requires getting rid of the boost iterator facade (or proposing and having a fix merged there)

**Reply by figueroa1395** [reply]
For sure, which I think is a good idea (dumping boost), albeit would requiere a lot of work. For this reason I am just dumping it here so we don't forget :)

**Comment by Jerry-Jinfeng-Guo** [comment]
Not sure if we should link the relevant issues with this thread, but here are they [in-complete, growing list]:
- https://github.com/PowerGridModel/power-grid-model/issues/809
- https://github.com/PowerGridModel/power-grid-model/issues/806
- https://github.com/PowerGridModel/power-grid-model/issues/798
- https://github.com/PowerGridModel/power-grid-model/issues/794
- https://github.com/PowerGridModel/power-grid-model/issues/791
- https://github.com/PowerGridModel/power-grid-model/issues/839

**Reply by Jerry-Jinfeng-Guo** [reply]
Functor class for certain cases where a local var is defined but need to be passed to a lambda via `std::move` or lambda is too complex

**Comment by nitbharambe** [comment]
@Jerry-Jinfeng-Guo  @figueroa1395  This discussion is specifically for next major version ideas. These are ideas introducing breaking change but minor enough to forget if we ever plan to do the bump in fae future.

**Reply by figueroa1395** [reply]
We shall then add these to a separate issue, where non-breaking changes that we still wish to do live, such that we don't forget.

**Comment by mgovers** [comment]
https://github.com/PowerGridModel/power-grid-model/issues/934
